skale/transactions/tools.py

```diff
@@ -32,24 +32,23 @@
 logger = logging.getLogger(__name__)
 
 
-def make_dry_run_call(wallet, method, gas_limit) -> dict:
+def make_dry_run_call(wallet, method) -> dict:
     opts = {
         'from': wallet.address,
-        'gas': gas_limit
     }
     logger.info(
         f'Dry run tx: {method.fn_name}, '
         f'sender: {wallet.address}, '
         f'wallet: {wallet.__class__.__name__}, '
-        f'gasLimit: {gas_limit}'
     )
     try:
-        call_result = method.call(opts)
+        estimated_gas = method.estimateGas(opts)
```

Comment:
if `gas_limit` - do `call(...)` instead and return limit_gas instead of estimated_gas